File: app/src/main/java/ru/coffeeplanter/androidlogin/data/settings/Crypter.java
package ru.coffeeplanter.androidlogin.data.settings;

import android.util.Base64;
import android.util.Log;

import javax.crypto.Cipher;
import javax.crypto.spec.SecretKeySpec;

/**
 * Cryption class.
 */

@SuppressWarnings("SpellCheckingInspection")
class Crypter {

    private final String TAG = "Crypter";

    private String mAlgorithm;

    private Cipher mEncipher;
    private Cipher mDecipher;

    @SuppressWarnings("WeakerAccess")
    Crypter(byte[] key, String algorithm) {
        try {
            this.mAlgorithm = algorithm;
            mEncipher = Cipher.getInstance(getAlgorithm());
            mDecipher = Cipher.getInstance(getAlgorithm());
            SecretKeySpec secretKeySpec = new SecretKeySpec(key, getAlgorithm());
            mEncipher.init(Cipher.ENCRYPT_MODE, secretKeySpec);
            mDecipher.init(Cipher.DECRYPT_MODE, secretKeySpec);
        }
        catch (Exception e) {
            Log.e(TAG, "Error while initializing Crypter object: ", e);
        }
    }

    Crypter(byte[] key) {
        this(key, "AES");
    }

    String encrypt(String str) {
        try {
            byte[] encrypted = mEncipher.doFinal(str.getBytes("UTF8"));
            return Base64.encodeToString(encrypted, Base64.DEFAULT);
        }
        catch (Exception e) {
            Log.e(TAG, "Encryption error: ", e);
        }
        return str;
    }

    String decrypt(String str) {
        try {
            byte[] decrypted = Base64.decode(str, Base64.DEFAULT);
            return new String(mDecipher.doFinal(decrypted), "UTF8");
        }
        catch (Exception e) {
            Log.e(TAG, "Decryption error: ", e);
        }
        return str;
    }

    @SuppressWarnings("WeakerAccess")
    String getAlgorithm() {
        return mAlgorithm;
    }

}

File: app/src/main/java/ru/coffeeplanter/androidlogin/data/settings/Crypter.kt
package ru.coffeeplanter.androidlogin.data.settings

import android.util.Base64
import android.util.Log

import javax.crypto.Cipher
import javax.crypto.spec.SecretKeySpec

/**
 * Cryption class.
 */

internal class Crypter(key: ByteArray, algorithm: String) {

    @Suppress("PrivatePropertyName")
    private val TAG = "Crypter"

    @Suppress("MemberVisibilityCanPrivate")
    var algorithm: String? = null
        private set

    private var mEncipher: Cipher? = null
    private var mDecipher: Cipher? = null

    init {
        try {
            this.algorithm = algorithm
            mEncipher = Cipher.getInstance(algorithm)
            mDecipher = Cipher.getInstance(algorithm)
            val secretKeySpec = SecretKeySpec(key, algorithm)
            mEncipher!!.init(Cipher.ENCRYPT_MODE, secretKeySpec)
            mDecipher!!.init(Cipher.DECRYPT_MODE, secretKeySpec)
        } catch (e: Exception) {
            Log.e(TAG, "Error while initializing Crypter object: ", e)
        }
    }

    constructor(key: ByteArray) : this(key, "AES")

    fun encrypt(str: String): String {
        try {
            val encrypted = mEncipher!!.doFinal(str.toByteArray(charset("UTF8")))
            return Base64.encodeToString(encrypted, Base64.DEFAULT)
        } catch (e: Exception) {
            Log.e(TAG, "Encryption error: ", e)
        }

        return str
    }

    fun decrypt(str: String): String {
        try {
            val decrypted = Base64.decode(str, Base64.DEFAULT)
            return String(mDecipher!!.doFinal(decrypted), charset("UTF8"))
        } catch (e: Exception) {
            Log.e(TAG, "Decryption error: ", e)
        }
        return str
    }

}
